### PR TITLE
Divertor target swap

### PR DESCRIPTION
**Pull request by DarioV86**
#432 

In some cases, depending on the target type (horizontal or vertical), the plasma flux region side and the scrape off layer side of the target appeared to be swapped.

Now it should work. Sorry @sebkahn, I didn't find any more elegant way.

### CODE REVIEW

**Comment by sebkahn** [comment]
Nice improvement of the code structure, this is way easier to understand now!

File: BLUEPRINT/systems/firstwall.py [hunk 1]
```diff
@@ -134,6 +134,11 @@ def get_tangent_vector(point_on_loop, loop):
 
     # Return normalised tangent vector
     tangent_norm = tangent / np.linalg.norm(tangent)
+
+    # Fixing the tangent vector direction to be concordant with the x-axis
+    if tangent_norm[0] > 0:
+        tangent_norm = -tangent_norm
```
Comment (sebkahn) [inline BLUEPRINT/systems/firstwall.py:140]: This is a nice way to do it !

File: BLUEPRINT/systems/firstwall.py [hunk 2]
```diff
@@ -520,10 +515,7 @@ def make_outer_leg(self, div_top_right, outer_strike, middle_point, flux_loop):
         tangent = get_tangent_vector(outer_strike, flux_loop)
 
         # Get the outer target points
-        (
-            outer_target_internal_point,
-            outer_target_external_point,
-        ) = self.make_divertor_target(
+        (outer_target_pfr_end, outer_target_sol_end) = self.make_divertor_target(
```
Comment (sebkahn) [inline BLUEPRINT/systems/firstwall.py:518]: Better name !

**Review by sebkahn — APPROVED** [review]

**Comment by sebkahn** [comment]
Merge pending for a correction on STEP reactor design!

File: examples/BLUEPRINT/base/base_tutorial.ipynb [hunk 3]
```diff
@@ -213,7 +213,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/base/base_tutorial.ipynb:216]: Can you revert this

File: examples/BLUEPRINT/beams/tour_eiffel.ipynb [hunk 4]
```diff
@@ -236,7 +236,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/beams/tour_eiffel.ipynb:239]: Can you revert this

File: examples/BLUEPRINT/cad/cad_mug_challenge.ipynb [hunk 5]
```diff
@@ -157,7 +157,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/cad/cad_mug_challenge.ipynb:160]: Can you revert this

File: examples/BLUEPRINT/cad/cad_tutorial.ipynb [hunk 6]
```diff
@@ -502,7 +502,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/cad/cad_tutorial.ipynb:505]: Can you revert this

File: examples/BLUEPRINT/cad/make_some_cad.ipynb [hunk 7]
```diff
@@ -96,7 +96,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/cad/make_some_cad.ipynb:99]: Can you revert this

File: examples/BLUEPRINT/geometry/geometry_tutorial.ipynb [hunk 8]
```diff
@@ -817,7 +817,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/geometry/geometry_tutorial.ipynb:820]: Can you revert this

File: examples/BLUEPRINT/nova/first_wall_example.ipynb [hunk 9]
```diff
@@ -403,7 +403,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/nova/first_wall_example.ipynb:406]: Can you revert this

File: examples/BLUEPRINT/nova/heat_flux_calculation.ipynb [hunk 10]
```diff
@@ -96,7 +96,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/nova/heat_flux_calculation.ipynb:99]: Can you revert this

File: examples/BLUEPRINT/nova/heat_flux_calculation_double_null.ipynb [hunk 11]
```diff
@@ -116,7 +116,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/nova/heat_flux_calculation_double_null.ipynb:119]: Can you revert this

File: examples/BLUEPRINT/nova/how_to_design_tfcoils.ipynb [hunk 12]
```diff
@@ -578,7 +578,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/nova/how_to_design_tfcoils.ipynb:581]: Can you revert this

File: examples/BLUEPRINT/nova/tf_cage_example.ipynb [hunk 13]
```diff
@@ -253,7 +253,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/nova/tf_cage_example.ipynb:256]: Can you revert this

File: examples/BLUEPRINT/systems/config_tutorial.ipynb [hunk 14]
```diff
@@ -640,7 +640,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/systems/config_tutorial.ipynb:643]: Can you revert this

File: examples/BLUEPRINT/systems/divertor_silhouette_tutorial.ipynb [hunk 15]
```diff
@@ -385,7 +385,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/BLUEPRINT/systems/divertor_silhouette_tutorial.ipynb:388]: Can you revert this

File: examples/design/EU-DEMO/EUDEMO_reactor.ipynb [hunk 16]
```diff
@@ -698,7 +698,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/design/EU-DEMO/EUDEMO_reactor.ipynb:701]: Can you revert this

File: examples/equilibria/eudemo_2017.ipynb [hunk 17]
```diff
@@ -377,7 +377,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/equilibria/eudemo_2017.ipynb:380]: Can you revert this

File: examples/equilibria/single_null.ipynb [hunk 18]
```diff
@@ -299,7 +299,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/equilibria/single_null.ipynb:302]: Can you revert this

File: examples/geometry/geometry_tutorial.ipynb [hunk 19]
```diff
@@ -448,7 +448,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/geometry/geometry_tutorial.ipynb:451]: Can you revert this

File: examples/materials/material_definitions.ipynb [hunk 20]
```diff
@@ -182,7 +182,7 @@
       "name": "python",
       "nbconvert_exporter": "python",
       "pygments_lexer": "ipython3",
-      "version": "3.8.12"
+      "version": "3.8.10"
```
Comment (sebkahn) [inline examples/materials/material_definitions.ipynb:185]: Can you revert this

**Review by sebkahn — DISMISSED** [review]

**Review by je-cook — APPROVED** [review]

**Pull request merged**